src/integer/mat_z/get.rs

```diff
@@ -95,6 +95,33 @@ impl GetEntry<Z> for MatZ {
     }
 }
 
+impl MatZ {
+    /// Efficiently collects all [`fmpz`]s in a [`MatZ`] without cloning them.
+    ///
+    /// # Example
+    /// ```compile_fail
+    /// use math::intger::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let mat = MatZ::from_str("[[1,2],[3,4],[5,6]]").unwrap();
+    ///
+    /// let fmpz_entries = mat.collect_entries();
+    /// ```
```

Comment:
Please add a hint for developers, that if the corresponding matrix is dropped, the vector of fmpz might suffer from memory leaks, since only the pointers are cloned, and the value behind the pointers might have changed

Reply:
Done

Reply:
Use Lifetimes